backend/users/tests.py

```diff
@@ -273,13 +274,77 @@ def test_patch_existing_avatar_not_logged(self):
         self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
 
     def test_patch_existing_avatar_logged_owner(self):
-        self.client.force_authenticate(user=self.test_user1)
+        self.client.force_authenticate(user=self.user_1)
         response = self.client.patch(self.url_detail, self.avatar_data)
 
         self.assertEqual(response.status_code, status.HTTP_200_OK)
 
     def test_patch_existing_avatar_logged_not_owner(self):
-        self.client.force_authenticate(user=self.test_user2)
+        self.client.force_authenticate(user=self.user_2)
         response = self.client.patch(self.url_detail, self.avatar_data)
 
         self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+
+class UserFavouritesTestCase(APITestCase):
+    @classmethod
+    def setUpTestData(cls):
+        cls.user_1 = UserFactory()
+        cls.url_add_favourite_presentations = reverse(
+            "users:user-add_to_favourite_presentations", kwargs={"pk": cls.user_1.id}
+        )
+        cls.url_add_favourite_tags = reverse(
+            "users:user-add_to_favourite_tags", kwargs={"pk": cls.user_1.id}
+        )
+        cls.url_remove_favourite_presentations = reverse(
+            "users:user-remove_from_favourite_presentations",
+            kwargs={"pk": cls.user_1.id},
+        )
+        cls.url_remove_favourite_tags = reverse(
+            "users:user-remove_from_favourite_tags", kwargs={"pk": cls.user_1.id}
+        )
+
+        cls.tag = TagFactory(name="Django")
+        cls.presentation = PresentationFactory()
+        cls.tag_data = {"favourite_tags": [cls.tag.id]}
+        cls.presentation_data = {"favourite_presentations": [cls.presentation.id]}
+
+    def test_add_favourite_tags(self):
+        self.client.force_authenticate(user=self.user_1)
+        response = self.client.put(self.url_add_favourite_tags, self.tag_data)
+
+        self.user_1.refresh_from_db()
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertTrue(self.tag in self.user_1.favourite_tags.all())
```

Comment:
Maybe Django is smart enough to optimize this, but in worst case this would download all tags, iterate over the collection and check if `self.tag` is there, which is suboptimal. If you can do something easily on DB-level you should do it instead of doing that in Python. So in this case you would do:

`self.assertTrue(self.user_1.favourite_tags.filter(id=sefl.tag.id).exists())`